src/internal/operators/withLatestFrom.ts

```diff
@@ -147,68 +146,65 @@ export function withLatestFrom<T, R>(array: ObservableInput<any>[], project: (..
  * each input Observable.
  * @name withLatestFrom
  */
-export function withLatestFrom<T, R>(...inputs: any[]): OperatorFunction<T, R> {
-  return (source: Observable<T>) => {
-    let project: (...values: any[]) => R;
-    if (typeof inputs[inputs.length - 1] === 'function') {
-      project = inputs.pop();
-    }
-
-    return lift(source, function (this: Subscriber<R | T[]>, source: Observable<T>) {
-      const subscriber = this;
-      const len = inputs.length;
-      const otherValues = new Array(len);
-      // An array of whether or not the other sources have emitted. Matched with them by index.
-      // TODO: At somepoint, we should investigate the performance implications here, and look
-      // into using a `Set()` and checking the `size` to see if we're ready.
-      let hasValue = inputs.map(() => false);
-      // Flipped true when we have at least one value from all other sources and
-      // we are ready to start emitting values.
-      let ready = false;
+export function withLatestFrom<T, R>(...inputs: any[]): OperatorFunction<T, R | any[]> {
+  let project: (...values: any[]) => R;
+  if (typeof inputs[inputs.length - 1] === 'function') {
+    project = inputs.pop();
```

Comment:
this fixes an issue that actually wasn't a bug _purely by coincidence_. Before this change, the arguments array was being mutated on every subscription to the result (via `pop()`). However, we're just lucky that the majority of the time, only the very last argument can be a function, so it would not mutate the second time around. (It's possible that this wouldn't be true if someone made a function that was also thennable or implemented `Symbol.observable` or something... but that's a pretty extreme corner case)... Either way, it was resolved when I moved the argument handling out of the initialization to accommodate the new `operate` function.